Parsing the flag can end up in a nil pointer if err == nil

https://github.com/gaia-pipeline/gaia/blob/master/server/server.go#L77

This code segment here can cause a nil pointer exception is err is nil in the call `err.Error()`.

Reproduce: 

Run a worker with `.gaia_config` file defined and no parameter:

```
vagrant@ubuntu-bionic:~/goproject/gaia$ ./main
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xbc7742]

goroutine 1 [running]:
github.com/gaia-pipeline/gaia/server.Start(0xc00014bf88, 0xc000086058)
        /home/vagrant/goproject/gaia/server/server.go:77 +0x92
main.main()
        /home/vagrant/goproject/gaia/cmd/gaia/main.go:11 +0x22
```

--- a/server/server.go
+++ b/server/server.go
@@ -74,8 +74,10 @@ func init() {
 // Start initiates all components of Gaia and starts the server/agent.
 func Start() (err error) {
 	// Parse command line flags
-	if err := fs.Parse(os.Args[1:]); err.Error() == "flag: help requested" {
-		return nil
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		if err.Error() == "flag: help requested" {
+			return nil
+		}
 	}
 
 	// Check version switch

